librz/include/rz_analysis.h

```diff
@@ -711,6 +718,14 @@ typedef struct rz_analysis_var_t {
 
 	// below members are just for caching, TODO: remove them and do it better
 	int argnum;
+
+	struct {
+		enum {
```

Comment:
Good choice. We will have other origins in the future - PDB, OMF. Maybe also decompiler-originated variables could be useful, e.g. from rz-ghidra

Reply:
Just please add a commentary for those

Reply:
I recommend to make origin a struct with a "kind" and "u" union fields, so that it can be extended in the future without breaking the API/ABI. But then, it is better to store "origin" as a pointer not as structure inside. @thestr4ng3r @wargio @ret2libc what do you think about that?